src/Capability/Registry.php

```diff
@@ -271,4 +272,57 @@ public function hasElements(): bool
             || !empty($this->prompts)
             || !empty($this->resourceTemplates);
     }
+
+    /**
+     * Export the current discovery state (only discovered elements, not manual ones).
+     */
+    public function exportDiscoveryState(): DiscoveryState
```

Comment:
if feel like `get` and `set` are more common than `export` and `import` for state
```suggestion
    public function getDiscoveryState(): DiscoveryState
```